CMakeLists.txt

```diff
@@ -92,20 +99,19 @@ option(FMT_OS "Include core requiring OS (Windows/Posix) " ON)
 option(FMT_MODULE "Build a module instead of a traditional library." OFF)
 option(FMT_SYSTEM_HEADERS "Expose headers with marking them as system." OFF)
 
-set(FMT_CAN_MODULE OFF)
-if (CMAKE_CXX_STANDARD GREATER 17 AND (
-  (MSVC AND MSVC_VERSION GREATER 1933) OR
-  ((CMAKE_CXX_COMPILER_ID MATCHES "Clang") AND
-   (CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL "16.0"))))
-  set(FMT_CAN_MODULE ON)
+set(FMT_CAN_MODULE ${FMT_MODULE})
+if (CMAKE_CXX_STANDARD GREATER 17)
+  if (MSVC_VERSION GREATER 1933 OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+    set(FMT_CAN_MODULE ON)
+  endif ()
 endif ()
```

Comment:
Der Versionscheck ist essentiell! Meiner Kenntnis nach funktioniert das ganze erst mit Clang 16.0 - und sei es nur wegen der Commandline Syntax. Momentan können nur MSVC (prinzipiell bereits ab msvc 19.29) und Clang 16. `FMT_CAN_MODULE` liefert die Aussage "ja, damit kannst du {fmt} als Module übersetzen".

Reply:
Das ist OK so, IMHO ist das lesbaren, und mit den () to complex!